Sources/Fluent/Query/QueryRepresentable.swift

```diff
@@ -96,12 +134,30 @@ extension QueryRepresentable {
         if let _ = model.id, model.exists {
             try model.willUpdate()
             var row = try model.makeRow()
-            if E.usesTimestamps {
+            try row.set(E.idKey, model.id)
+
+            // timestampable
+            if
+                let T = E.self as? Timestampable.Type,
+                let t = model as? Timestampable
+            {
                 let now = Date()
-                try row.set(E.updatedAtKey, now)
-                model.storage.updatedAt = now
+                try row.set(T.updatedAtKey, now)
+                t.updatedAt = now
             }
-            try row.set(E.idKey, model.id)
+
+            // soft deletable
+            if
+                let S = E.self as? SoftDeletable.Type,
+                let s = model as? SoftDeletable
+            {
+                if let deletedAt = s.deletedAt {
+                    try row.set(S.deletedAtKey, deletedAt)
+                } else {
+                    try row.set(S.deletedAtKey, Node.null)
```

Comment:
Not sure I follow here, so converting `nil` to `null` internally on all optionals? Might be possible, but I think it will have some unexpected results, let me see what's possible and maybe we can make it easy to give an option for users who prefer this.